### PR TITLE
Support LazyFrames in `plotting.plot_panels`

### LINKED ISSUE #74: `plotting.plot_panel` fails with LazyFrames
This code results in the following error: `AttributeError: 'LazyFrame' object has no attribute 'get_column'`

```python
import polars as pl

y = pl.scan_parquet("https://github.com/descendant-ai/functime/raw/main/data/commodities.parquet")

plotting.plot_panel(y)
```

This is due to `get_column` being available for `pl.DataFrames` only. I think `plot_panel` could benefit from lazy operations since only number of entities * `last_n` points are plotted. The following line would just require adding a `.collect()` in the end if `y` is a `LazyFrame`.

https://github.com/neocortexdb/functime/blob/e02509c47882f556640b7c88299988cfbdde8836/functime/plotting.py#L54

### PR DESCRIPTION
**Pull request by baggiponte**
Fixes #74 and adds a couple of type annotations.

### CODE REVIEW

**Comment by baggiponte** [comment]
With the latest commit ([ef807f5](https://github.com/neocortexdb/functime/pull/79/commits/ef807f593e1b944853b69b2c581994654a5ecb6c)) I also fix #80

**Comment by baggiponte** [comment]
@topher-lo would you mind reviewing?

File: functime/plotting.py [hunk 1]
```diff
@@ -115,7 +122,7 @@ def plot_forecasts(
     """
 
     # Get most recent observations
-    entity_col, time_col, target_col = y_true.columns
+    entity_col, time_col, target_col = y_true.columns[:2]
```
Comment (topher-lo) [inline functime/plotting.py:125]:
```suggestion
    entity_col, time_col, target_col = y_true.columns[:3]
```

**Review by topher-lo — CHANGES_REQUESTED** [review]
List indexing error.

File: functime/plotting.py (hunk 1 above)
Comment (baggiponte) [reply functime/plotting.py:125]: oh yes you are right, :3 takes cols 0 1 and 2.

**Comment by baggiponte** [comment]
Can't explain the conflicts, will open a new PR.

**Pull request closed without merging**